test/e2e/main_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	k8sclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func TestMain(m *testing.M) {
+	// e2e test job does not guarantee our operator is up before
+	// launching the test, so we need to do so.
+	err := waitForOperator()
+	if err != nil {
+		fmt.Printf("failed waiting for operator to start: %v\n", err)
+		os.Exit(1)
+	}
+	os.Exit(m.Run())
+}
+
+func waitForOperator() error {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return err
+	}
+	kclient, err := k8sclient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
@bparees any particular reason you're not using the same pattern as in https://github.com/openshift/cluster-image-registry-operator/blob/1500c6432360d2eb0328ab4e537ab0c3b9214207/test/e2e/main_test.go#L41-L61 ?

Reply:
if you're asking about the client, it's because that uses the operator SDK for the client and this repo doesn't use the operator SDK.

Reply:
Shouldn't we be consistent across the board and have similar patterns?

Reply:
@soltysh and someday maybe we will.  Today we have operators written in many different ways, vendoring many different libraries, so the tests reflect that.